### PR TITLE
octoprint: assorted plugin maintenance

### PR DESCRIPTION
**Pull request by lovesegfault**
###### Motivation for this change
This was done while I was working on #98917, but didn't fit in that PR. I noticed a couple of the octoprint plugins were out of date, so I bumped them, and in the process added a new plugin that I need.

All were tested and work.

###### Things done

- [x] Tested using sandboxing ([nix.useSandbox](https://nixos.org/nixos/manual/options.html#opt-nix.useSandbox) on NixOS, or option `sandbox` in [`nix.conf`](https://nixos.org/nix/manual/#sec-conf-file) on non-NixOS linux)
- Built on platform(s)
   - [x] NixOS
   - [ ] macOS
   - [ ] other Linux distributions
- [ ] Tested via one or more NixOS test(s) if existing and applicable for the change (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
- [ ] Tested compilation of all pkgs that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review wip"`
- [x] Tested execution of all binary files (usually in `./result/bin/`)
- [x] Determined the impact on package closure size (by running `nix path-info -S` before and after)
- [x] Ensured that relevant documentation is up to date
- [x] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/.github/CONTRIBUTING.md).

### CODE REVIEW

File: pkgs/applications/misc/octoprint/plugins.nix [hunk 1]
```diff
@@ -272,15 +272,34 @@ in {
...
+
+    src = fetchFromGitHub {
+      owner = "AliceGrey";
+      repo = "OctoprintKlipperPlugin";
+      rev = version;
+      sha256 = "15yg2blbgqp2gdpsqqm8qiiznq5qaq8wss07jimkl0865vrvlz7l";
+    };
+
+    meta = with stdenv.lib; {
+      description = "A plugin for a better integration of Klipper into OctoPrint.";
```
Comment (IvarWithoutBones) [inline pkgs/applications/misc/octoprint/plugins.nix:287]: `meta.description` should not end with a period as per the [contributing guidelines](https://github.com/NixOS/nixpkgs/blob/master/.github/CONTRIBUTING.md#submitting-changes).

**Review by IvarWithoutBones — CHANGES_REQUESTED** [review]
LGTM except for 1 minor problem.

**Comment by j0hax** [comment]
Thanks for updating so quickly.

Unfortunately my previous commits (#99010) have been open for over a week now.

File: pkgs/applications/misc/octoprint/plugins.nix (hunk 1 above)
Comment (lovesegfault) [reply pkgs/applications/misc/octoprint/plugins.nix:287]: Fixed :)

**Comment by lovesegfault** [comment]
cc. @andir for a final review / landing

**Comment by gebner** [comment]
Thanks!  Evaluation error is unrelated.

**Pull request merged**